parquet/src/main/java/org/apache/iceberg/parquet/TypeWithSchemaVisitor.java

```diff
@@ -54,101 +57,108 @@ public static <T> T visit(
     } else {
       // if not a primitive, the typeId must be a group
       GroupType group = type.asGroupType();
-      OriginalType annotation = group.getOriginalType();
```

Comment:
The existing code was refactored into methods for each case. Handling for LIST and MAP did not change.